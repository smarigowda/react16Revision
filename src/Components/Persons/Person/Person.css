.Person {
    width: 30%;
    margin: auto;
    margin-top: 30px;    
    margin-bottom: 30px;
    padding: 20px;
    border: 1px solid #eee;
    box-shadow: 0 2px 3px rgb(20, 19, 19);
    text-align: center;
}

@media (min-width: 500px) {
    .Person {
        width: 450px;
    }
}