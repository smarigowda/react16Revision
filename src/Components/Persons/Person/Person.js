import React, { Component } from 'react';
import classes from './Person.css';
import WithClass from '../../../hoc/WithClass';

class Person extends Component {
    constructor(props) {
        super(props);
        console.log('[Person.js] constructor', props);
    }

    componentWillMount() {
        console.log('[Person.js] componentWillMount');
    }

    componentDidMount() {
        console.log('[Person.js] componentDidMount');
        if( this.props.position === 0 ) {
            this.inputElement.focus();
        }
    }
    render() {
        console.log('[Person.js] render()');
        return (
            <WithClass classes={classes.Person}>
                <p id="person" onClick={this.props.click}>I'm {this.props.name}, I am {this.props.age} years old</p>
                <p id="message">{this.props.children}</p>
                <input
                    ref={ ele => {
                        this.inputElement = ele;
                    }}
                    type="text"
                    onChange={this.props.changeHandler}
                    value={this.props.name}
                />
            </WithClass>
        ) 
    }
}

export default Person;