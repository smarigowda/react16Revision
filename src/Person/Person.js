import React from 'react';
import classes from './Person.css';

const person = props => {
    return (
        <div className={classes.Person}>
            <p id="person" onClick={props.click}>I'm {props.name}, I am {props.age} years old</p>
            <p id="message">{props.children}</p>
            <input type="text" onChange={props.changeHandler} value={props.name}/>
        </div>
    )
}

export default person;