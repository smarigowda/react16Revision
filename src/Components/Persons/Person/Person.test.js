import React from 'react';
import ReactDOM from 'react-dom';
import Person from './Person';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Person />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('shallow renders without crashing', () => {
  shallow(<Person />);
});

it('renders a paragraph with name and age', () => {
  const wrapper = shallow(<Person name="Santosh" age="45"/>);
  // const expectedMatching = expect.stringMatching(/I'm \w+, I am \d+ years old/);
  const expectedMatching = expect.stringMatching(/I'm Santosh, I am 45 years old/);
  expect(wrapper.text()).toEqual(expectedMatching);
  expect(wrapper.find('#person')).toHaveLength(1);
});

it('renders the text inside the tag', () => {
  const wrapper = shallow(<Person name="Santosh" age="44">My Hobby: Coding</Person>);
  const expectedMatching = expect.stringMatching(/My Hobby: Coding/);
  expect(wrapper.text()).toEqual(expectedMatching);
});

it('renders correctly', () => {
  const clickHandler = jest.fn();
  const tree = renderer.create(<Person name="Santosh" age="45" click={clickHandler} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Person should have on click handler', () => {
  const clickHandler = jest.fn();
  const wrapper = shallow(<Person click={clickHandler}/>);
  expect(wrapper.find('#person').props()).toHaveProperty('onClick');
  expect(wrapper.find('#person').prop('onClick')).toBeDefined();
});

it('Clicking on the Person should call the event handler function', () => {
  const clickHandler = jest.fn();
  const wrapper = mount(<Person click={clickHandler}/>);
  wrapper.find('#person').simulate('click');
  expect(clickHandler.mock.calls.length).toBe(1);
});

it('Person should have an input element', () => {
  const wrapper = shallow(<Person name="Santosh" age="45"/>);
  expect(wrapper.find('input')).toHaveLength(1);
});

it('Input element should have a onChange attr', () => {
  const wrapper = shallow(<Person name="Santosh" age="45"/>);
  expect(wrapper.find('input').props()).toHaveProperty('onChange');
});

it('When the input value is changed, change handler should be called', () => {
  const changeHandler = jest.fn();
  const wrapper = mount(<Person changeHandler={changeHandler}/>);
  wrapper.find('input').simulate('change', {target: {value: 'My new value'}});
  expect(changeHandler.mock.calls.length).toBe(1);
})

