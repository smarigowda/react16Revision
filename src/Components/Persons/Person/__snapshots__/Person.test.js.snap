// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Person"
>
  <p
    id="person"
    onClick={[Function]}
  >
    I'm 
    Santosh
    , I am 
    45
     years old
  </p>
  <p
    id="message"
  />
  <input
    onChange={undefined}
    type="text"
    value="Santosh"
  />
</div>
`;
